cranelift/assembler-x64/meta/src/instructions/bitmanip.rs

```diff
@@ -39,5 +39,8 @@ pub fn list() -> Vec<Inst> {
         inst("cwtd", fmt("ZO", [w(implicit(dx)), r(implicit(ax))]), rex([0x66, 0x99]), _64b | compat),
         inst("cltd", fmt("ZO", [w(implicit(edx)), r(implicit(eax))]), rex([0x99]), _64b | compat),
         inst("cqto", fmt("ZO", [w(implicit(rdx)), r(implicit(rax))]), rex([0x99]).w(), _64b),
+
+        inst("bswapl", fmt("O", [rw(r32)]), rex([0x0F, 0xC8]).rd(), _64b | compat),
+        inst("bswapq", fmt("O", [rw(r64)]), rex([0x0F, 0xC8]).w().ro(), _64b),
```

Comment:
Ok question on this one for you: for the 32-bit variant the Intel manual has `0F C8+rd ` so `.rd()` makes sense above for `bswapl`, but the 64-bit version has `REX.W + 0F C8+rd ` in the manual. I've switched that to `.ro()` here given the assertions added in #10832 though. While I know pdf-search isn't great I wasn't able to actually find any instructions using `+ro`, but I could find some with `+rd` and `+rw` (also no hits with `+rb`).

Basically wanted to flag this, I'm not sure if it's something where we should relax the assertions here or if I'm missing something?

Reply:
I saw that too and my take is that it is a typo in the manual. I'll flag it internally but I'm pretty confident that using `ro` is more consistent.